ndcube/extra_coords/extra_coords.py

```diff
@@ -405,6 +419,111 @@ def dropped_world_dimensions(self):
+            A new ExtraCoords object holding the interpolated coords.
+
+        """
+        new_ec = type(self)(ndcube)
+        if self.is_empty:
+            return new_ec
+        cube_dims = self._ndcube.dimensions.value.astype(int)
+        naxes = len(cube_dims)
+        if len(new_array_grids) != naxes:
+            raise ValueError("new_array_grids must have an entry for each array axis."
+                             f"num. array axes: {naxes}; "
+                             f"num. new array grids: {len(new_array_grids)}")
+        array_order_mapping = convert_between_array_and_pixel_axes(np.asarray(self.mapping), naxes)
+        old_array_grids = [None if new_grid is None else np.arange(dim)
+                           for new_grid, dim in zip(new_array_grids, cube_dims)]
+        # For SkyCoords, two keys must be done at once.
+        # Define list for tracking which keys should be skipped for this reason.
+        skip = []
+        i = -1
+        for key in self.keys():
+            # If a key has already been handled because it is linked to a SkyCoord
+            # also linked to a previous key, skip it.
+            # This implementation assumes there is only one SkyCoord extra coord.
+            if key in skip:
+                continue
+            # Counter must be incremented here as there can be more keys than lookup tables.
+            # Skipped keys point to a table already handled by a previous iteration.
+            # So only move to next lookup table (i.e. increment i) if key is not skipped.
+            i += 1
+            aom = array_order_mapping[i]
+            lut = self._lookup_tables[i]
+            if new_array_grids[aom] is None:
+                new_coord = lut[1]
+                name = key
+            else:
+                table = lut[1].table
+                if isinstance(table, tuple):
+                    table = table[0]
+                if isinstance(table, SkyCoord):
+                    # Special-case SkyCoord as even when 1-D it has 2 components, lon and lat.
+                    # Find other key corresponding to this SkyCoord.
+                    sky_key = None
+                    remaining_keys = list(self.keys())[i:]
+                    j = 0
+                    while not sky_key:
+                        j += 1
+                        next_key = remaining_keys[j]
+                        if isinstance(self[next_key]._lookup_tables[0][1],
+                                      SkyCoordTableCoordinate):
+                            sky_key = next_key
+                    # Interpolate lon and lat components separately then recombine into a SkyCoord.
+                    celestial_names = dict(
+                        [(item, key)
+                         for key, item in table.representation_component_names.items()])
+                    lon = getattr(table, celestial_names["lon"])
+                    lat = getattr(table, celestial_names["lat"]).to(lon.unit)
```

Comment:
not all SkyCoords are lon / lat and not all of them are 2D.

Reply:
Any issue with this as a solution?
```python
new_components = []
for name in table.representation_component_names.keys():
    component = getattr(table, name).value
    comp_interp = scipy.interpolate.interp1d(old_array_grids[aom], component)
    new_components.append(comp_interp(new_array_grids[aom]))
new_coord = SkyCoord(*new_components, unit=table.representation_component_units.values(),
                     frame=table.frame)
```